messaging/src/main/java/org/axonframework/commandhandling/gateway/CommandGateway.java

```diff
@@ -55,7 +56,8 @@
      * Sends the given {@code command} and wait for it to execute. The result of the execution is returned when
      * available. This method will block indefinitely, until a result is available, or until the Thread is interrupted.
      * When the thread is interrupted, this method returns {@code null}. If command execution resulted in an exception,
-     * it is wrapped in a {@link CommandExecutionException}.
+     * it is wrapped in a {@link CommandExecutionException}. If command dispatching failed,
+     * {@link CommandDispatchException} is thrown instead.
```

Comment:
I am thinking of adding "the `CommandDispatchException` is only thrown in a distributed environment". 
So, something like this:

`If command dispatching failed when in a distributed environment, a {@link CommandDispatchException} is thrown instead.`

What do you think?

Reply:
Note that dispatching doesn't require a distributed environment to fail. There might be other conditions in which dispatching fails, e.g. when queuing for async processing, but the queue is full.
I wouldn't be too specific about conditions in which things can happen on the interface level.

Reply:
Fair point.
That would opt to move the exception out of the `distributed` package as soon as we can, correct?